src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget())
+                .thenReturn(MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L,  SelectionMode.CHEAPEST));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final MultiBuyCustomLineItemsTarget newTarget =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE);
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithDifferentCustomMaxOccurrence_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget())
+                .thenReturn(MultiBuyCustomLineItemsTarget
+                        .of("quantity > 0", 6L, 3L,  SelectionMode.MOST_EXPENSIVE, 1L));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final MultiBuyCustomLineItemsTarget newTarget =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE, 2L);
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithSameMultiBuyCustomLineItemsTargetValues_ShouldNotBuildUpdateAction() {
+        // Given a cart with 6 items, the discount can be applied only once.
+        // As a result, 3 custom items will be discounted
+        // and 4 most expensive custom items will be marked as participating in this discount.
+        final MultiBuyCustomLineItemsTarget target =
+                MultiBuyCustomLineItemsTarget.of("quantity > 0", 6L, 3L, SelectionMode.MOST_EXPENSIVE);
+
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget()).thenReturn(target);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getTarget()).thenReturn(target);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithJustNewTargetHasNullValue_ShouldBuildUpdateAction() {
```

Comment:
`WithOnlyNullNewTarget`

Reply:
why is this test not there anymore?

Reply:
according to the API, this is not possible, but both can be null in case of GiftLineItem. If the old is not null, new cannot be un-setted.